recipes/pagmo2/all/conanfile.py

```diff
@@ -0,0 +1,145 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class Pagmo2Conan(ConanFile):
+    name = "pagmo2"
+    description = "pagmo is a C++ scientific library for massively parallel optimization."
+    license = ("LGPL-3.0-or-later", "GPL-3.0-or-later")
+    topics = ("conan", "pagmo", "optimization", "parallel-computing", "genetic-algorithm", "metaheuristics")
+    homepage = "https://esa.github.io/pagmo2"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_eigen": [True, False],
+        "with_nlopt": [True, False],
+        "with_ipopt": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_eigen": False,
+        "with_nlopt": False,
+        "with_ipopt": False,
+    }
+
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15.7",
+            "gcc": "7",
+            "clang": "5.0",
+            "apple-clang": "9.1"
+        }
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 17)
+
+        def lazy_lt_semver(v1, v2):
+            lv1 = [int(v) for v in v1.split(".")]
+            lv2 = [int(v) for v in v2.split(".")]
+            min_length = min(len(lv1), len(lv2))
+            return lv1[:min_length] < lv2[:min_length]
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn("{} {} requires C++17. Your compiler is unknown. Assuming it supports C++17.".format(self.name, self.version))
+        elif lazy_lt_semver(str(self.settings.compiler.version), minimum_version):
+            raise ConanInvalidConfiguration("{} {} requires C++17, which your compiler does not support.".format(self.name, self.version))
+
+    def requirements(self):
+        self.requires("boost/1.76.0")
+        self.requires("tbb/2020.3")
+        if self.options.with_eigen:
+            self.requires("eigen/3.3.9")
+        if self.options.with_nlopt:
+            self.requires("nlopt/2.7.0")
+        if self.options.with_ipopt:
+            raise ConanInvalidConfiguration("ipopt recipe not available yet in CCI")
```

Comment:
```suggestion
```

Reply:
Probably harmless, but not a good idea I think: we should prevent the dependency graph to be resolved if we already know that it doesn't make sense (here a missing dependency).

Reply:
done